foreign_cc/cmake.bzl

```diff
@@ -73,23 +72,136 @@ def _create_configure_script(configureParameters):
     flags = get_flags_info(ctx, "<TARGET>")
     no_toolchain_file = ctx.attr.cache_entries.get("CMAKE_TOOLCHAIN_FILE") or not ctx.attr.generate_crosstool_file
 
+    cmake_commands = []
+
+    # If the legacy `make_commands` attribute was not set, use the new
+    # `targets` api for building our target.
+    if not ctx.attr.make_commands:
+        data = ctx.attr.data + getattr(ctx.attr, "tools_deps", [])
+        configuration = "Debug" if is_debug_mode(ctx) else "Release"
+
+        # Generate a list of arguments for cmake's build command
+        build_args = " ".join([
+            ctx.expand_location(arg, data)
+            for arg in ctx.attr.build_args
+        ])
+
+        # Generate commands for all the targets, ensuring there's
+        # always at least 1 call to the default target.
+        for target in ctx.attr.targets or [""]:
+            # Note that even though directory is always passed, the
+            # following arguments can take precedence.
+            cmake_commands.append("{cmake} --build {dir} --config {config} {args} {target}".format(
+                cmake = attrs.cmake_path,
+                dir = ".",
+                args = build_args,
+                target = target,
+                config = configuration,
+            ))
+
+        if ctx.attr.install:
+            # Generate a list of arguments for cmake's install command
+            install_args = " ".join([
+                ctx.expand_location(arg, data)
+                for arg in ctx.attr.install_args
+            ])
+
+            cmake_commands.append("{cmake} --install {dir} --config {config} {args}".format(
+                cmake = attrs.cmake_path,
+                dir = ".",
+                args = install_args,
+                config = configuration,
+            ))
+
     define_install_prefix = "export INSTALL_PREFIX=\"" + _get_install_prefix(ctx) + "\"\n"
     configure_script = create_cmake_script(
         workspace_name = ctx.workspace_name,
-        cmake_path = configureParameters.attrs.cmake_path,
+        generator = attrs.generator,
+        cmake_path = attrs.cmake_path,
         tools = tools,
         flags = flags,
         install_prefix = "$$INSTALL_PREFIX$$",
         root = root,
         no_toolchain_file = no_toolchain_file,
         user_cache = dict(ctx.attr.cache_entries),
-        user_env = dict(ctx.attr.env_vars),
-        options = ctx.attr.cmake_options,
+        user_env = getattr(ctx.attr, "env_vars", {}),
+        options = attrs.generator_args,
```

Comment:
I'm not convinced that `generator_args` is the correct name for what is essentially setting the cmake cache variables from the command line. 
I envisaged having a `generator` attr and keeping the existing `cmake_options` and then appending the `generator` value with `-G` to the values in `cmake_options`. We can present a migration warning if `-G` is passed in `cmake_options` and is a generator we explicitly support?

Reply:
Thinking through it some more I think I'm fine with it as is. I'd confused `cmake_options` and `cache_entries` a bit in my thinking.